bool is_red
bool is_blue